spec/contract.spec.json

```diff
@@ -0,0 +1,105 @@
+{
+  "id": "truffle-contract-schema.json",
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "properties": {
+    "contractName": { "$ref": "#/definitions/ContractName" },
+    "abi": { "$ref": "#/definitions/ABI" },
+    "bytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "Bytecode sent as contract-creation transaction data" }
+    ]},
+    "deployedBytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "On-chain deployed contract bytecode" }
+    ]},
+    "sourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract-creation transaction data bytecode" }
+    ]},
+    "deployedSourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract bytecode" }
+    ]},
+    "source": { "$ref": "#/definitions/Source" },
+    "sourcePath": { "$ref": "#/definitions/SourcePath" },
+    "ast": { "$ref": "#/definitions/AST" },
+    "address": { "$ref": "#/definitions/Address" },
+    "networks": {
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": { "$ref": "#/definitions/Network" }
+      },
+      "additionalProperties": false
+    },
+    "schemaVersion": { "$ref": "#/definitions/SchemaVersion" },
+    "updatedAt": {
+      "type": "string",
+      "format": "date-time"
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+  ],
+  "definitions": {
+    "ContractName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]*$"
+    },
+    "Address": {
+      "type": "string",
+      "pattern": "^0x[a-fA-F0-9]{40}$"
+    },
+    "ABI": {
+      "type": "array"
+    },
+    "Bytecode": {
+      "type": "string",
+      "pattern": "^0x0$|^0x([a-fA-F0-9]{2}|__[a-zA-Z0-9_]{38})+$"
+    },
+    "Source": {
+      "type": "string"
+    },
+    "SourceMap": {
+      "type": "string"
+    },
+    "SourcePath": {
+      "type": "string"
+    },
+    "AST": {
+      "type": "object"
+    },
+    "Network": {
+      "type": "object",
+      "properties": {
+        "address": { "$ref": "#/definitions/Address" },
+        "links": {
+          "type": "object",
+          "patternProperties": {
+            "^[a-zA-Z_][a-zA-Z0-9_]*$": { "$ref": "#/definitions/Link" }
+          },
+          "additionalProperties": false
+        }
+      }
+    },
+    "Link": {
+      "type": "object",
+      "properties": {
+        "address": { "$ref": "#/definitions/Address" },
+        "events": {
+          "type": "object",
+          "patternProperties": {
+            "^0x[a-fA-F0-9]{64}$": { "$ref": "#/definitions/Event" }
+          },
+          "additionalProperties": false
+        }
+      }
+    },
+    "Event": {
+      "type": "object"
```

Comment:
I wouldn't, see above. Although you *could* copy the event portion of the abi json schema since we're using that within our own. I think either way (either copying the abi's schema for events, or doing nothing) is good for now.